apps/ejabberd/src/mongoose_privacy.erl

```diff
@@ -0,0 +1,66 @@
+%%%-------------------------------------------------------------------
+%%% @author bartek <bartlomiej.gorny@erlang-solutions.com>
+%%% @copyright (C) 2017, Erlang Solutions sp. z o.o.
+%%% @doc
+%%%
+%%% @end
+%%% Created : 06. Feb 2017 11:37
+%%%-------------------------------------------------------------------
+-module(mongoose_privacy).
+-author("bartek").
+
+-include("ejabberd.hrl").
+-include("mod_privacy.hrl").
+
+%% API
+-export([privacy_check_packet/6, privacy_check_packet/7]).
+
+%%% API %%%
+
+-spec privacy_check_packet(Acc :: mongoose_acc:t(),
```

Comment:
In the function spec you used record:
```erlang
PrivacyList :: #userlist{}
```

Elvis doesn't like it and I agree with him. It's better have a type for this record and use the type.